torch/csrc/utils/pythoncapi_compat.h

```diff
@@ -0,0 +1,575 @@
+// Header file providing new C API functions to old Python versions.
+//
+// File distributed under the Zero Clause BSD (0BSD) license.
+// Copyright Contributors to the pythoncapi_compat project.
+//
+// Homepage:
+// https://github.com/python/pythoncapi_compat
+//
+// Latest version:
+// https://raw.githubusercontent.com/python/pythoncapi_compat/master/pythoncapi_compat.h
```

Comment:
Did not review this. Maybe we should submodule it?

Reply:
I was thinking about it but in their readme, they recommend to vendor it. So that's what I did.
I can move this to be a submodule if anyone has a strong feeling about it!

Reply:
Ok if they say vendor, vendor it